silkworm/node/stagedsync/forks/main_chain.cpp

```diff
@@ -281,7 +281,7 @@ bool MainChain::notify_fork_choice_update(Hash head_block_hash, std::optional<Ha
 
         const auto finalized_block_number = get_block_number(*finalized_block_hash);
         last_finalized_head_.number = *finalized_block_number;
-        last_finalized_head_.hash = std::move(*finalized_block_hash);
+        last_finalized_head_.hash = *finalized_block_hash;
```

Comment:
This is another example where I do not like the clang-tidy rules: it is true that _at the moment_ finalized_block_hash; is trivially-copyable so we do not need std::move but what happens if someone changes the type of finalized_block_hash making it non trivially-copyable? it will be very difficult to find all the places where a non-trivial copy is made and it is better to do a move. 
So I consider "performance defensive programming"  to code the above line as before:

`last_finalized_head_.hash = std::move(*finalized_block_hash);`

it is not an error, it states that we no longer need finalized_block_hash and it can be moved. It will be a performance gain in a future where finalized_block_hash is not trivially copyable.

Reply:
Oh, in that case there's another clang-tidy check that'd suggest to use an `std::move`. So the answer is more clang-tidy, not less ;)

Reply:
this might convince me ;-)